user-interface/src/lib/components/uswds/ButtonGroup.test.tsx

```diff
@@ -0,0 +1,135 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import ButtonGroup, { ButtonGroupProps } from './ButtonGroup';
+import Button from './Button';
+
+const buttonGroupId = 'test-button-group';
+const buttonGroupCSSClass = 'test-class-name';
+
+describe('tests for USWDS button group', () => {
+  function renderWithProps(props?: Partial<ButtonGroupProps>) {
+    const defaultProps: ButtonGroupProps = {
+      id: buttonGroupId,
+      activeButtonId: 'button-1',
+      className: buttonGroupCSSClass,
+      onButtonClick: (_id: string) => {},
+      children: [],
+    };
+
+    const renderProps = { ...defaultProps, ...props };
+    render(
+      <div>
+        <ButtonGroup {...renderProps} />
+      </div>,
+    );
+  }
+
+  test('should contain supplied class name and id, and render without children if no children are passed', () => {
+    renderWithProps();
+
+    const ul = screen.getByTestId(`button-group-${buttonGroupId}`);
```

Comment:
What is the significance of the name `ul`? All I can think is unordered list, but that doesn't seem to be what the test id is referring to.